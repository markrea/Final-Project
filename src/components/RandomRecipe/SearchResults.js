import React from "react";
import PropTypes from "prop-types";
import "../../styles/SearchResults.css";

const Result = ({ results }) => {
  return (
    <div key={results.image} className="innerSearchResult">
      <img src={results.image} alt="mockAlt" className="card-photo" />
      <figcaption className="caption">{results.title}</figcaption>
      <a href={results.link} target="_blank" rel="noopener noreferrer">
        View Recipe
      </a>
    </div>
  );
};
const SearchResults = ({ searchResults }) => {
  return (
    <div className="SearchResults" data-testid="results-div">
      <div className="test">
        {searchResults.map((results) => (
          <Result key={results.id} results={results} />
        ))}
      </div>
    </div>
  );
};

Result.propTypes = {
  results: PropTypes.objectOf.isRequired,
};
SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf.isRequired,
};

export default SearchResults;
