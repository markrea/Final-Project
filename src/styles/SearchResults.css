.innerSearchResult {
    display: inline-grid;
    grid-column: repeat(3, 1fr);
    margin-left: 5%;
    margin-right: 3%;
    margin-top: 3rem;
    border: 1px dotted rgba(105, 168, 2, 0.555);
    width: 20%;
    justify-content: center;

}

.card-photo{
        object-fit: cover;
       
}

.caption{
    text-align: center;
    padding: 1rem;
    color: rgb(67, 88, 7);
    font-size: 1rem;
    width: 90%;
}
