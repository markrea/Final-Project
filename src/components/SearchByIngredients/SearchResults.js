import React, { useState, useEffect } from "react";
import axios from "axios";
import PropTypes from "prop-types";
import "../../styles/SearchResults.css";

const API_KEY2 = "48aba1bc5bce4ba3901fc4b1e3591949";
const Result = ({ results }) => {
  const [href, setHref] = useState("");
  useEffect(() => {
    axios
      .get(
        `https://api.spoonacular.com/recipes/${results.id}/information?apiKey=${API_KEY2}`
      )
      .then((response) => {
        const link = response.data.sourceUrl;
        return setHref(link);
      })
      .catch((err) => console.log(err));
  }, []);
  return (
    <div key={results.image} className="innerSearchResult">
      <img src={results.image} alt="mockAlt" className="card-photo" />
      <figcaption className="caption">{results.title}</figcaption>
      <a href={href} target="_blank" rel="noopener noreferrer" >
        More Details
      </a>
    </div>
  );
};
const SearchResults = ({ searchResults }) => {
  return (
    <div className="SearchResults" data-testid="results-div">
      <div className="test">
        {searchResults.map((results) => (
          <Result key={results.id} results={results} />
        ))}
      </div>
    </div>
  );
};
SearchResults.propTypes = {
  Result: PropTypes.array,
};
export default SearchResults;
