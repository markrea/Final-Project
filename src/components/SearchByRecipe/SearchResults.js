import React from "react";
import PropTypes from "prop-types";
import "../../styles/SearchResults.css";

const Result = ({ results }) => {
  const calories = Math.round(results.calories);
  return (
    
      <div key={results.image} className="innerSearchResult">
        <img src={results.image} alt="mockAlt" className="card-photo" />
        <figcaption className="caption">
          <b>{results.title}</b>
        </figcaption>
        <figcaption className="caption">
          Ready in {results.readyInMinutes}
          mins
        </figcaption>
        <figcaption className="caption">
          Serves
          {results.serves}
        </figcaption>
        <figcaption className="caption"> {calories} Calories</figcaption>
        <a href={results.link} target="_blank" rel="noopener noreferrer">
          View Recipe
        </a>
      </div>
   
  );
};
const SearchResults = ({ searchResults }) => {
  if (searchResults[0]) {
    return (
      <div className="SearchResults" data-testid="results-div">
        <div className="test">
          {searchResults.map((results) => (
            <Result key={results.id} results={results} />
          ))}
        </div>
      </div>
    );
  }
  return (
    <div className="noResults">
      <p>No recipes found. Please try another search</p>
    </div>
  );
};
SearchResults.propTypes = {
  Result: PropTypes.array,
};
export default SearchResults;
