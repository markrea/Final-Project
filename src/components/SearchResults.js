import React from "react";
import PropTypes from "prop-types";

import "../styles/SearchResults.css";
import recipeLink from "../requests/recipeLink";


const SearchResults = ({ searchResults }) => {
  return (
    <div className="SearchResults" data-testid="results-div">
      <div className="test">
        {searchResults.map((results) => {
          return (
            <div key={results.image} className="innerSearchResult">
              <img src={results.image} alt="mockAlt" className="card-photo" />

              <figcaption className="caption">{results.title}</figcaption>
              <a href={recipeLink(results.id)}>More Details</a>
              {console.log(recipeLink(results.id))}
            </div>
          );
        })}
      </div>
    </div>
  );
};
SearchResults.propTypes = {
  searchResults: PropTypes.array,
};

export default SearchResults;
