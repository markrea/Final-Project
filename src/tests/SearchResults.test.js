import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";
import "@testing-library/jest-dom/extend-expect";

const props = {
  searchResults: [],
};

describe("SearchResults", () => {
  xit("component renders with correct props", () => {
    const { asFragment } = render(<SearchResults {...props} />);
    expect(asFragment).toMatchSnapshot();
  });
  xit("returns a search results div", () => {
    const { getByTestId } = render(<SearchResults {...props} />);
    const searchResultsDiv = getByTestId("results-div");
    expect(searchResultsDiv).toBeInTheDocument();
  });
  xit("returns an image for each search result", () => {
    const { getByAltText } = render(<SearchResults {...props} />);
    const img = getByAltText("mockAlt");
    expect(img).toBeInTheDocument(); // not working
  });
});
