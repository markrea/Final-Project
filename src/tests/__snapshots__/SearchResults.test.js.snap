// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders with correct props 1`] = `[Function]`;
